paper-input-container.d.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * DO NOT EDIT
+ *
+ * This file was automatically generated by
+ *   https://github.com/Polymer/gen-typescript-declarations
+ *
+ * To modify these typings, edit the source file(s):
+ *   paper-input-container.html
+ */
+
+/// <reference path="../polymer/types/polymer.d.ts" />
+/// <reference path="../iron-flex-layout/iron-flex-layout.d.ts" />
+/// <reference path="../paper-styles/default-theme.d.ts" />
+/// <reference path="../paper-styles/typography.d.ts" />
+
+interface PaperInputContainerElement extends Polymer.Element {
+
+  /**
+   * Set to true to disable the floating label. The label disappears when the input value is
+   * not null.
+   */
+  noLabelFloat: boolean|null|undefined;
+
+  /**
+   * Set to true to always float the floating label.
+   */
+  alwaysFloatLabel: boolean|null|undefined;
+
+  /**
+   * The attribute to listen for value changes on.
+   */
+  attrForValue: string|null|undefined;
+
+  /**
+   * Set to true to auto-validate the input value when it changes.
+   */
+  autoValidate: boolean|null|undefined;
+
+  /**
+   * True if the input is invalid. This property is set automatically when the input value
+   * changes if auto-validating, or when the `iron-input-validate` event is heard from a child.
+   */
+  invalid: boolean|null|undefined;
+
+  /**
+   * True if the input has focus.
+   */
+  readonly focused: boolean|null|undefined;
+  _addons: any[]|null|undefined;
+  _inputHasContent: boolean|null|undefined;
+  _inputSelector: string|null|undefined;
+  _boundOnFocus: Function|null|undefined;
+  _boundOnBlur: Function|null|undefined;
+  _boundOnInput: Function|null|undefined;
+  _boundValueChanged: Function|null|undefined;
+  readonly _valueChangedEvent: any;
+  readonly _propertyForValue: any;
+  readonly _inputElement: any;
+  readonly _inputElementValue: any;
+  ready(): void;
+  attached(): void;
+  _onAddonAttached(event: any): void;
+  _onFocus(): void;
+  _onBlur(): void;
+  _onInput(event: any): void;
+  _onValueChanged(event: any): void;
+  _handleValue(inputElement: any): void;
+  _handleValueAndAutoValidate(inputElement: any): void;
+  _onIronInputValidate(event: any): void;
+  _invalidChanged(): void;
```

Comment:
Lots of methods which should be private I suppose? We can remove these later.

Reply:
Done (some of them anyway).